django/db/models/sql/query.py

```diff
@@ -306,7 +306,8 @@ def clone(self):
         obj.annotations = self.annotations.copy()
         if self.annotation_select_mask is not None:
             obj.annotation_select_mask = self.annotation_select_mask.copy()
-        obj.combined_queries = tuple(query.clone() for query in self.combined_queries)
+        if self.combined_queries:
+            obj.combined_queries = tuple(query.clone() for query in self.combined_queries)
```

Comment:
We can also go faster by using a list comprehension instead of a generator - they're cheaper to create:

```suggestion
            obj.combined_queries = tuple([query.clone() for query in self.combined_queries])
```

Simple benchmark:

```
In [2]: items = [1, 2, 3, 4]

In [3]: %timeit tuple(x*2 for x in items)
508 ns ± 10 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [4]: %timeit tuple([x*2 for x in items])
358 ns ± 14.7 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

Reply:
I agree with Adam. @kezabelle I can push edits and merge it before the feature freeze, if you agree with the proposed change.

Reply:
@felixxm yep, I agree with it, I was just going to avoid doing it until after the freeze so you'd have less to do :)